main.py

```diff
@@ -523,6 +523,9 @@ def ui_access_wrapper(self, *args, **kwargs):
     get_redirect_route(
         r'%s/' % feconf.SUGGESTION_URL_PREFIX,
         suggestion.SuggestionHandler),
+    get_redirect_route(
```

Comment:
I'm not aware of this but can you explain why we are not handling this on activity level like we have the structure for accepting and rejecting actions? 

![image](https://user-images.githubusercontent.com/16653571/56147561-288cf300-5fc6-11e9-8a73-20e704746c4f.png)


is there any issue with this pattern?

Reply:
@import-keshav I mean for actions like "Accept" and "Reject" we have a separate handler for each activity right? If yes, then why we are not having this pattern, is there any issue with this? (I'm not aware of this, It would be great if you can explain this.)

Reply:
I am not sure about the issue but @nithusha21 has suggested me to make edit suggestion handler more generalized.

Reply:
That's a good point @DubeySandeep. It is currently split up for every activity type only for accept and rejects. (This is because the permissions for the handlers for accepting and reject ing are activity dependent). For general actions like edits/resubmission, we have separate controllers (This is because the permission checks aren't really activity dependent in this case). So I think here it is fine to have a separate controller.

Reply:
ping @DubeySandeep

Reply:
I agree with you Nithesh but I see that we are doing lots of permission check in the edit acl too (See this [comment](https://github.com/oppia/oppia/pull/6368#discussion_r277132300)). 

@import-keshav  Can you please check whether moving this to the activity-dependent handlers would make this simple?